src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -325,6 +329,72 @@ EnsureCitusTableCanBeCreated(Oid relationOid)
 }
 
 
+/*
+ * undistribute_table gets a distributed table name and 
+ * udistributes the table.
+ */
+Datum
+undistribute_table(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+
+	Oid relationId = PG_GETARG_OID(0);
+	const char *relationName = get_rel_name(relationId);
+
+	EnsureTableOwner(relationId);
+
+	Relation relation = relation_open(relationId, ExclusiveLock);
+
+	StringInfo query = makeStringInfo();
+
+	SPI_connect();
+
+	List * tableDDLEvents = GetTableDDLEvents(relationId, true);
+	char * createTableQuery = linitial(tableDDLEvents);
+	list_delete_first(tableDDLEvents);
+
+	Node *parseTree = ParseTreeNode(createTableQuery);
+
+	((CreateStmt *)parseTree)->relation->relname = "tmp1";
+
+	CitusProcessUtility(parseTree, createTableQuery, PROCESS_UTILITY_TOPLEVEL, NULL, None_Receiver, NULL);
+
+	resetStringInfo(query);
+	appendStringInfo(query, "INSERT INTO tmp1 SELECT * FROM %s", relationName);
+	SPI_execute(query->data, false, 0);
+
+	ObjectAddress undistributeTableAddres;
+
+	undistributeTableAddres.classId = RelationRelationId;
+	undistributeTableAddres.objectId = relationId;
+	undistributeTableAddres.objectSubId = 0;
+
+	DirectFunctionCall3(master_drop_all_shards,
+						ObjectIdGetDatum(relationId),
+						CStringGetTextDatum(relationName),
+						CStringGetTextDatum("public"));
```

Comment:
We should use the same schema as the table is in, not `"public"` (everywhere in the change).